.github/workflows/test-and-build-docker-images.yml

```diff
@@ -63,21 +63,45 @@ jobs:
       uses: actions/setup-go@v3
       with:
         go-version: ${{ inputs.go-version }}
+        cache: true
+        cache-dependency-path: '${{ inputs.working-dir }}go.sum'
 
     - name: Set up gotestsum
       if: inputs.artifact != 'bridge2'
       run: go install gotest.tools/gotestsum@v1.8.1
 
+    - name: Download Go dependencies
+      if: inputs.artifact != 'bridge2'
+      working-directory: ./${{ inputs.artifact }}
+      run: |
+        go mod download
+
     - name: Test ${{ inputs.artifact }} # TODO add --race #8486 
       if: inputs.artifact != 'bridge2' && ((inputs.build-everything == 'true') || (inputs.should-run == 'true'))
       working-directory: ./${{ inputs.artifact }}
       run: |
-        go mod download
         gotestsum --no-color=false --format=testname -- -coverprofile=coverage.txt -covermode=atomic -v ./... 
 
     #######################################################################
           # TESTS FOR BRIDGE
     #######################################################################
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      if: inputs.artifact == 'bridge2'
+      working-directory: bridge
+      run: |
+        echo "::set-output name=dir::$(yarn cache dir)"
+    
+    - name: Setup yarn cache
+      uses: actions/cache@v2
+      id: yarn-cache
+      if: inputs.artifact == 'bridge2'
+      with:
+        path: |
+          **/node_modules
+          ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ inputs.docker-test-target }}-${{ hashFiles('**/yarn.lock') }}
```

Comment:
Since we only have one Node based artifact anyways, why not either fully parametrize it and use the working dir of the bridge artifact, or fully hard code it with the paths from bridge?